src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java

```diff
@@ -677,10 +677,17 @@ private static boolean isWithinIPv4OctetRange(int number) {
 
     public static Map<String, Set<String>> getComputerTags(Computer computer) {
         Set<LabelAtom> labels = null;
-        try {
-            labels = computer.getNode().getAssignedLabels();
-        } catch (NullPointerException e) {
-            logger.fine("Could not retrieve labels");
+        assert computer != null : "Computer is unexpectedly null";
+        var node = computer.getNode();
+        if (node != null) {
+            Set<LabelAtom> assignedLabels = node.getAssignedLabels();
+            if (assignedLabels != null) {
+                labels = node.getAssignedLabels();
+            } else {
+                logger.warning("Could not retrieve labels");
```

Comment:
Why is the log level being changed? It looks like we're risking spamming the users' logs

Reply:
It was [initially suggested](https://github.com/jenkinsci/datadog-plugin/pull/404/files/ca807d6f52df700cb89bf7d59308d39d7f50463e#r1554177118) to switch to a warning, but she said that she will defer to you, so I will just change it back to a fine.